sdk/samples/iot/aziot_esp32/aziot_esp32.ino

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <cstdlib>
+#include <string.h>
+#include <time.h>
+
+#include <WiFi.h>
+#include <mqtt_client.h>
+
+#include <az_iot_hub_client.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include "AzIoTSasToken.h"
+#include "SerialLogger.h"
+#include "ca.h"
+#include "iot_configs.h"
+
+#define sizeofarray(a) (sizeof(a) / sizeof(a[0]))
+#define NTP_SERVERS "pool.ntp.org", "time.nist.gov"
+#define QOS1 1
+#define DO_NOT_RETAIN_MSG 0
+#define SAS_TOKEN_DURATION_IN_MINUTES 60
+#define UNIX_TIME_NOV_13_2017 1510592825
+
+static const char* ssid = IOT_CONFIG_WIFI_SSID;
+static const char* password = IOT_CONFIG_WIFI_PASSWORD;
+static const char* host = IOT_CONFIG_IOTHUB_FQDN;
+static const char* mqtt_broker_uri = "mqtts://" IOT_CONFIG_IOTHUB_FQDN;
+static const char* device_id = IOT_CONFIG_DEVICE_ID;
+static const int port = 8883;
+
+static esp_mqtt_client_handle_t mqtt_client;
+static az_iot_hub_client client;
+
+static char mqtt_client_id[128];
+static char mqtt_username[128];
+static char mqtt_password[200];
+static uint8_t sas_signature_buffer[256];
+static unsigned long next_telemetry_send_time_ms = 0;
+static char telemetry_topic[128];
+static uint8_t telemetry_payload[100];
+static uint32_t telemetry_send_count = 0;
+
+static AzIoTSasToken sasToken(
+    &client,
+    AZ_SPAN_FROM_STR(IOT_CONFIG_DEVICE_KEY),
+    AZ_SPAN_FROM_BUFFER(sas_signature_buffer),
+    AZ_SPAN_FROM_BUFFER(mqtt_password));
+
+static void connectToWiFi()
+{
+  Logger.Info("Connecting to WIFI SSID " + String(ssid));
+
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(ssid, password);
+  while (WiFi.status() != WL_CONNECTED)
+  {
+    delay(500);
+    Serial.print(".");
+  }
+
+  Serial.println("");
+
+  Logger.Info("WiFi connected, IP address: " + WiFi.localIP().toString());
+}
+
+static void initializeTime()
+{
+  Logger.Info("Setting time using SNTP");
+
+  configTime(-5 * 3600, 0, NTP_SERVERS);
```

Comment:
what is -5?

Reply:
These are supposed to be GMT offsets, but kept like that due to c&p. I'll expand it.